operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/rate/PeriodRateLimiter.java

```diff
@@ -36,11 +30,12 @@ public PeriodRateLimiter(Duration refreshPeriod, int limitForPeriod) {
   }
 
   @Override
-  public Optional<Duration> acquirePermission(ResourceID resourceID) {
-    if (limitForPeriod <= 0) {
+  public Optional<Duration> isLimited(RateLimitState rateLimitState) {
```

Comment:
This is essentially where having a marker interface helps for documentation purposes as the alternative would be: `isLimited(Object state)` which wouldn't be as explicit. It's not a big deal but I think it's nicer this way.